core/shared/src/main/scala/zio/prelude/AssociativeFlatten.scala

```diff
@@ -240,4 +240,73 @@ trait AssociativeFlattenSyntax {
     def flatMap[B](f: A => F[B])(implicit flatten: AssociativeFlatten[F], covariant: Covariant[F]): F[B] =
       flatten.flatten(covariant.map(f)(fa))
   }
+
+  /**
+   * Provides infix syntax for filtering covariant types.
+   */
+  implicit class AssociativeFlattenCovariantIdentityBothIdentityEitherOps[F[+_], A](fa: F[A]) {
```

Comment:
this class is composition of several typeclasses, but why do you add this class in here?
I would like to know if there are any rules for this.